src/main/webapp/js/userMap.js

```diff
@@ -90,22 +104,106 @@ function handleData(err, userData) {
+                });
+            }
+        }
+    });
+
+    map.addPlugin('pins', function (layer, data, options) {
+        var self = this,
+            fillData = this.options.fills,
+            svg = this.svg;
+
+        if (!data || (data && !data.slice)) {
+            handleError();
+            return;
+        }
+
+        var markers = layer.selectAll('image.datamaps-pins').data(data, JSON.stringify);
+
+        markers
+        .enter()
+        .append('image')
+        .attr('class', 'datamaps-pin')
+        .attr('xlink:href', 'images/pin.png')
+        .attr('height', 20)
+        .attr('width', 20)
+        .attr('x', getX)
+        .attr('y', getY)
+        .on('mouseover', function (datum) {
+            var $this = d3.select(this);
+
+            if (options.popupOnHover) {
+                self.updatePopup($this, datum, options, svg);
+            }
+        })
+        .on('mouseout', function (datum) {
+            var $this = d3.select(this);
+
+            if (options.highlightOnHover) {
+                var previousAttributes = JSON.parse($this.attr('data-previousAttributes'));
+                for (var attr in previousAttributes) {
+                    $this.style(attr, previousAttributes[attr]);
+                }
             }
+            d3.selectAll('.datamaps-hoverover').style('display', 'none');
+        });
+
+        markers.exit()
+        .transition()
+        .delay(options.exitDelay)
+        .attr('height', 0)
+        .remove();
+
+        function getCoordinates(datum) {
+            var latLng;
+            if (datumHasCoords(datum)) {
+                latLng = self.latLngToXY(datum.latitude, datum.longitude);
+            } else if (datum.centered) {
+                latLng = self.path.centroid(svg.select('path.' + datum.centered).data()[0]);
+            }
+
+            return latLng;
+        }
+
+        function getX(datum) {
+            return getCoordinates(datum)[0];
+        }
+
+        function getY(datum) {
+            return getCoordinates(datum)[1];
         }
+
+        function datumHasCoords (datum) {
+            return typeof datum !== 'undefined' && typeof datum.latitude !== 'undefined' && typeof datum.longitude !== 'undefined';
+        }
+    });
+
+    map.pins(pins, {
+        popupOnHover: true,
+        popupTemplate: getTooltipContent
     });
 }
 
+function getTooltipContent(data) {
+    return '<div class="hoverinfo">'
+            + '<p>'
+                + '<span class="bold">'
+                + data.name
+                + '</span>'
+                + '<br>'
+                + 'Institution(s): '
+                + data.numOfInstitutions
+            + '</p>'
+         + '</div>';
+}
+
 document.addEventListener('DOMContentLoaded', function(event) {
-    d3.json('/js/userMapData.json', handleData);
-});
+    d3.json('/js/countryCoordinates.json', function(data){
+        countryCoordinates = data;
+        d3.json('/js/userMapData.json', handleData);
```

Comment:
What I was trying to say is, there should be a way to prevent the need of the global var `countryCoordinates`. From what I understand, if `handleData` is a function taking in 2 params, you should be able to make it like this:
```javascript
d3.json('/js/userMapData.json', function(err, userData) {
    handleData(err, countryCoordinates, userData);
    // or: handleData(err, data, userData);
});
```